src/server/service/parking-lot.ts

```diff
@@ -21,30 +20,41 @@ export interface IParkingLotService {
 
 export default class ParkingLotService implements IParkingLotService {
   async create(dto: CreateParkingLotDTO): Promise<ParkingLot> {
-    const parkingLot = new ParkingLotModel(dto);
+    const parkingLotRepository = getRepository(ParkingLot);
+    const parkingLot = new ParkingLot();
 
-    await parkingLot.save();
+    parkingLot.name = dto.name;
 
-    return parkingLot;
+    const createdParkingLot = await parkingLotRepository.save(parkingLot);
+
+    return createdParkingLot;
   }
 
   async update(dto: UpdateParkingLotDTO): Promise<ParkingLot> {
+    const parkingLotRepository = getRepository(ParkingLot);
     const parkingLot = await this.findById(dto.id);
 
-    await parkingLot.update(dto);
+    parkingLot.name = dto.name;
+
+    await parkingLotRepository.update(dto.id, parkingLot);
 
     return parkingLot;
   }
 
   async remove(id: string): Promise<ParkingLot> {
+    const parkingLotRepository = getRepository(ParkingLot);
     const parkingLot = await this.findById(id);
-    const removed = await parkingLot.remove();
 
-    return removed;
+    const parkingLotRemoved = await parkingLotRepository.remove(parkingLot, {});
+
+    return parkingLotRemoved;
   }
```

Comment:
Theres 2 available methods you can use to perform "Deletion" in TypeORM. One is `remove` the other is `delete`.

- `remove` removes an entity or array of entities
- `delete` removes records from the database, not relying on having the entity present

Here we have the `Id` from the request already, there's no need to first find the entity ourselves and then use `remove` to delete such entity from database.

Having the ID is just enough to have a single consumption to the database to achieve our requirement, which is deleting the ParkingLot with the given ID.

Reply:
Use the Repository API reference to use the right methods, TypeORM offers a lot of options to achieve a single goal, its up to you to implement the most performant and maintainable ones.

https://github.com/typeorm/typeorm/blob/master/docs/repository-api.md#repository-api

Reply:
I suggest doing:

```typescript
async remove(id: string): Promise<ParkingLot> {
    const parkingLotRepository = getRepository(ParkingLot);
    const deleted = await parkingLotRepository.delete({ id });

    return deleted;
}
```

Reply:
Thanks a lot! But due `delete` method return a `DeleteResult` i think that first, i should check if exists using `this.findById`, and save the data returned in a variable to return the parkingLot/Slot deleted.